server/src/main/java/org/apache/druid/server/coordinator/rules/IntervalLoadRule.java

```diff
@@ -38,15 +38,23 @@ public class IntervalLoadRule extends LoadRule
 
   private final Interval interval;
   private final Map<String, Integer> tieredReplicants;
+  private final boolean allowEmptyTieredReplicants;
 
   @JsonCreator
   public IntervalLoadRule(
       @JsonProperty("interval") Interval interval,
-      @JsonProperty("tieredReplicants") Map<String, Integer> tieredReplicants
+      @JsonProperty("tieredReplicants") Map<String, Integer> tieredReplicants,
+      @JsonProperty("allowEmptyTieredReplicants") Boolean allowEmptyTieredReplicants
   )
   {
-    this.tieredReplicants = tieredReplicants == null ? ImmutableMap.of(DruidServer.DEFAULT_TIER, DruidServer.DEFAULT_NUM_REPLICANTS) : tieredReplicants;
-    validateTieredReplicants(this.tieredReplicants);
+    this.allowEmptyTieredReplicants = allowEmptyTieredReplicants != null && allowEmptyTieredReplicants;
+
+    if (!this.allowEmptyTieredReplicants) {
+      this.tieredReplicants = tieredReplicants == null ? ImmutableMap.of(DruidServer.DEFAULT_TIER, DruidServer.DEFAULT_NUM_REPLICANTS) : tieredReplicants;
+    } else {
+      this.tieredReplicants = tieredReplicants == null ? ImmutableMap.of() : tieredReplicants;
+    }
+    validateTieredReplicants(this.tieredReplicants, this.allowEmptyTieredReplicants);
```

Comment:
Refactored this into a separate function instead, since `validateAndGetTieredReplicants` was starting to do two different things. Please let me know if this looks good.